applications/asset_tracker_v2/doc/sensor_module.rst

```diff
@@ -0,0 +1,110 @@
+
++------------------+-----------------+
+| Sensor data type | External device |
++==================+=================+
+| Humidity         | `BME680`_       |
++------------------+-----------------+
+| Temperature      | `BME680`_       |
++------------------+-----------------+
+| Acceleration     | `ADXL362`_      |
++------------------+-----------------+
+
+The module controls and collects data from the sensors by interacting with their :ref:`device drivers <device_model_api>` via :ref:`Zephyr's generic sensor API <sensor_api>`.
+
+Data sampling
+=============
+
+Whenever the module receives an :c:enum:`APP_EVT_DATA_GET` event and the :c:enum:`APP_DATA_ENVIRONMENTAL` type is present in the ``app_data`` list carried in the event, it will sample data.
+When data sampling has been carried out, the :c:enum:`SENSOR_EVT_ENVIRONMENTAL_DATA_READY` event is sent from the module containing the sampled environmental sensor values.
+
+.. note::
+   The nRF9160 DK does not have any external sensors.
+   If the sensor module is queried for sensor data when building for the DK, the event :enum:`SENSOR_EVT_ENVIRONMENTAL_NOT_SUPPORTED` is sent out by the module
+   upon data sampling.
+
+Motion detection
+================
+
+Motion is detected whenever acceleration in either X, Y or Z plane exceeds the configured threshold value.
+The threshold is set in one of two ways:
+
+1. When receiving the :c:enum:`DATA_EVT_CONFIG_INIT` event after boot. This event contains the default threshold value set by
+   :ref:`CONFIG_DATA_ACCELEROMETER_THRESHOLD` or retrieved from flash.
+#. When receiving the :c:enum:`DATA_EVT_CONFIG_READY` event. This occurs whenever a new threshold value has been updated from cloud.
+
+Both events contains an accelerometer threshold value ``accelerometer_threshold`` in m/s2, present in the event structure.
+
+Motion detection in the module is enabled and disabled in turn to control the amount of :c:enum:`SENSOR_EVT_MOVEMENT_READY` events that are sent out by the sensor module.
+This functionality acts as a sort of flow control to prevent the sensor module from spamming the rest of the application with events if the device is accelerating over
+the threshold for extended periods of time.
+
+This behavior is controlled by the applicaton module via the :c:enum:`APP_EVT_ACTIVITY_DETECTION_ENABLE` and :c:enum:`APP_EVT_ACTIVITY_DETECTION_DISABLE` events.
+The sensor module will only send out an :c:enum:`SENSOR_EVT_MOVEMENT_READY` event if movement is detected and activity detection is enabled.
+
+.. note::
+   The DK does not have an external accelerometer. However, button 2 on the DK can be used to trigger movement for testing purposes.
+
+Configuration options
+*********************
+
+.. _CONFIG_SENSOR_THREAD_STACK_SIZE:
+
+CONFIG_SENSOR_THREAD_STACK_SIZE - Sensor module thread stack size
+   This option increases the sensor module's internal thread stack size.
```

Comment:
Suggest to change `increases` -> `configures` as it can be adjusted in any direction